### PR TITLE
Keepalive

### PR DESCRIPTION
**Pull request by mikehelmick**
## Proposed Changes

* Add optional DB keepalive pings

**Release Note**

```release-note
New environment variable.
* `DB_MAX_CONN_IDLE_TIME` a duration, when set will be passed to DB connection pool. Default value of `1m`
```

### CODE REVIEW

File: pkg/database/database.go [hunk 1]
```diff
@@ -130,6 +137,10 @@ func (db *Database) OpenWithCacher(ctx context.Context, cacher cache.Cacher) err
 		return fmt.Errorf("database gorm.Open: %w", err)
 	}
 
+	if c.MaxConnectionIdleTime > 0 {
```
Comment (sethvargo) [inline pkg/database/database.go:140]: I haven't done as much research here, but wouldn't `SetConnMaxLifetime` work better (and avoid all this overhead)?

```go
db.DB().SetConnMaxLifetime(1*time.Minute)
```

Comment (mikehelmick) [reply pkg/database/database.go:133]: made that the default - we can start there and see if it helps

**Comment by mikehelmick** [comment]
just idle closes at first

**Pull request merged**